gstd/gstd_daemon.c

```diff
@@ -45,15 +47,23 @@ static gchar *gstd_daemon_get_pid_filename (const gchar * filename);
 static gboolean _initialized = FALSE;
 static gchar *pid_filename = NULL;
 
-void
+gboolean
 gstd_daemon_init (gint argc, gchar * argv[], gchar * pidfilename)
 {
   const gchar *process_name;
+  gchar *pid_path;
 
-  g_return_if_fail (argv);
+  g_return_val_if_fail (argv, FALSE);
 
   if (_initialized)
-    return;
+    goto out;
+
+  /* Check if pid dir available */
+  pid_path = gstd_daemon_get_pid_filename (pidfilename);
+  if (0 != g_access (pid_path, W_OK)) {
+    g_printerr ("Unable to access Gstd pid dir: %s\n", g_strerror (errno));
```

Comment:
Maybe, you can move this error message to `error` label below.

Reply:
it was, it has been removed.